sentry/sentry.go

```diff
@@ -0,0 +1,214 @@
+	}
+
+	for _, option := range options {
+		option(l)
+	}
+
+	// Default to live sentry packet capturer
+	if l.Capturer == nil {
+		client, err := raven.New(dsn)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create sentry client")
+		}
+
+		l.Capturer = &NonBlockingCapturer{Client: client}
+	}
+
+	return l, nil
+}
+
+// WithCapturer allows to specify which packet capturer should be used.
+func WithCapturer(c Capturer) Option {
+	return func(l *Logger) {
+		l.Capturer = c
+	}
+}
+
+// MinLevel provides a minimum level threshold, above which Sentry packtes will be triggered
+func MinLevel(level zap.Level) Option {
+	return func(l *Logger) {
+		l.minLevel = level
+	}
+}
+
+// TraceEnabled allows to change wether (the somewhat expensive, relatively)
+// stack trace generation takes place.
+func TraceEnabled(enabled bool) Option {
+	return func(l *Logger) {
+		l.traceEnabled = enabled
+	}
+}
+
+// TraceCfg allows to change the number of skipped frames, number of context lines and
+// list of go prefixes that are considered "in-app", i.e. "github.com/uber-go/zap".
+func TraceCfg(skip, context int, prefixes []string) Option {
+	return func(l *Logger) {
+		l.traceSkipFrames = skip
+		l.traceContextLines = context
+		l.traceAppPrefixes = prefixes
+	}
+}
+
+// Extra stores additional information for each Sentry event.
+func Extra(extra map[string]interface{}) Option {
+	return func(l *Logger) {
+		l.extra = extra
+	}
+}
+
+// Log sends Sentry information provided minimum threshold is met.
+func (l *Logger) Log(lvl zap.Level, msg string, fields ...zap.Field) {
+	l.log(lvl, msg, fields)
+}
+
+// Debug is a nop.
+func (l *Logger) Debug(msg string, fields ...zap.Field) {}
+
+// Info sends Sentry information provided minimum threshold is met.
+func (l *Logger) Info(msg string, fields ...zap.Field) {
+	l.log(zap.InfoLevel, msg, fields)
+}
+
+// Warn sends Sentry information provided minimum threshold is met.
+func (l *Logger) Warn(msg string, fields ...zap.Field) {
+	l.log(zap.WarnLevel, msg, fields)
+}
+
+// Error sends Sentry information provided minimum threshold is met.
+func (l *Logger) Error(msg string, fields ...zap.Field) {
+	l.log(zap.ErrorLevel, msg, fields)
+}
+
+// Panic sends Sentry information provided minimum threshold is met, then panics.
+func (l *Logger) Panic(msg string, fields ...zap.Field) {
+	l.log(zap.PanicLevel, msg, fields)
+	panic(msg)
+}
+
+// Fatal sends Sentry information provided minimum threshold is met, then exits.
+func (l *Logger) Fatal(msg string, fields ...zap.Field) {
+	l.log(zap.FatalLevel, msg, fields)
+	os.Exit(1)
+}
+
+// DFatal is a nop.
+func (l *Logger) DFatal(msg string, fields ...zap.Field) {}
+
+// With returns Sentry logger with additional context.
+func (l *Logger) With(fields ...zap.Field) zap.Logger {
+	// Consider pre-allocing the whole new resulting map, but that may be even more expensive
+	for _, f := range fields {
+		f.AddTo(l.extra)
+	}
+	return l
```

Comment:
Most of the time, we log at `InfoLevel` and below; in those cases, any work done in the Sentry logger is pure overhead, since we typically use Sentry only for errors and above.